sql/sql_class.h

```diff
@@ -1601,13 +1601,14 @@ class Bloom_filter {
   ~Bloom_filter()
   {
     // Do not delete, just destruct, due to MEM_ROOT allocation
-    bit_set->~bitset();
+    if (bit_set != NULL)
+      bit_set->~bitset();
```

Comment:
Now I'm wondering whether it was correct for me to substitute delete with destructor call in the first place. The std::bitset apparently has no destructor. Perhaps it should be clear() here?

Reply:
That's why it was accidentally working (but I think it was still UB) - it was possible that bit_set was null at this point.

I don't see the point of calling clear here - the next variables reusing the same place in memory will reinitialize anyway, and this Bloom_filter is destroyed at this point.

Reply:
Since std::bitset destructor is default, then nothing has to be done here, and Bloom_filter destructor may default as well?